### PR TITLE
Creating Xmatters transport

### PR DESCRIPTION
**Pull request by LoveSkylark**
Transport to communicate with Xmatters.com using API

Small modifications are done to the payload.

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [ ] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [ ] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: LibreNMS/Alert/Xmatters.php [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            'priority' => $this->mapSeverityToPriority($alert_data['severity']),
+        ];
+
+        $data['properties']['event_state'] = $this->mapStateToEventState($alert_data['state']);
+        $data['properties']['instance_url'] = $device_url;
+        $data['properties']['device_url'] = ($device_url . '/device/' . $alert_data['device_id']);
+        $data['properties']['alert_url'] = ($device_url . '/device/' . $alert_data['device_id'] . '/alerts/' . $alert_data['alert_id']);
+        $data['properties']['hostname'] = $alert_data['sysName'];
+        $data['properties']['ip'] = $alert_data['hostname'];
+        $data['properties']['device_groups'] = $device_groups;
```
Comment (murrant) [inline LibreNMS/Alert/Xmatters.php:38]: These feel like custom fields applicable to your install.  Are they generically applicable?

File: LibreNMS/Alert/Xmatters.php [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    {
+        $url = $this->config['xmatters_url'] . '/api/integration/1/functions/' . $this->config['xmatters_function'] . '/triggers';
+
+        $device_url = \Config::get('base_url');
+        $device_groups = \DeviceCache::get($alert_data['device_id'])->groups->pluck('name');
+
+        // Build request data
+        $data = [
+            'properties' => $alert_data,
+            'recipients' => $alert_data['contacts'],
```
Comment (murrant) [inline LibreNMS/Alert/Xmatters.php:28]: The contacts field is suspect imo.  It will likely be removed in the future as nothing else is using it.

**Comment by murrant** [comment]
@LoveSkylark still working on this?

File: LibreNMS/Alert/Xmatters.php (hunk 2 above)
Comment (LoveSkylark) [reply LibreNMS/Alert/Xmatters.php:28]: Under stood

**Comment by LoveSkylark** [comment]
> @LoveSkylark still working on this?
Not really, I suspect there aren't and many xmatters users here that would benefit from this and I have this setup and working in my Cluster deployment script so I'm ok with just dropping this if this is in anyway "iffy"

**Comment by PipoCanaja** [comment]
Ok. We can close this for now, and if you want to get back to it, feel free to re-open. 
Thx

**Pull request closed without merging**